sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/MostFrequent.java

```diff
@@ -0,0 +1,747 @@
+            c.output(KV.of(pair.getKey().getElement(), pair.getValue()));
+          }
+        }
+      };
+    }
+
+    /**
+     * {@link DoFn} that retrieves the top k most frequent elements in a
+     * {@link PCollection} of {@link StreamSummarySketch} containing serializable objects.
+     *
+     * @param k     the top k most frequent elements to return
+     */
+    public static <T> DoFn<StreamSummarySketch<T>, KV<T, Long>> globallySerializable(int k) {
+      return new DoFn<StreamSummarySketch<T>, KV<T, Long>>() {
+        @ProcessElement
+        public void apply(ProcessContext c) {
+          List<KV<T, Long>> top = c.element().estimateTopK(k);
+          for (KV<T, Long> pair : top) {
+            c.output(KV.of(pair.getKey(), pair.getValue()));
+          }
+        }
+      };
+    }
+
+    /**
+     * {@link DoFn} that retrieves the top k most frequent elements per key in a
+     * {@link PCollection} of {@link KV}s with {@link StreamSummarySketch} containing
+     * non serializable objects wrapped in {@link ElementWrapper}.
+     *
+     * @param k     the top k most frequent elements to return
+     */
+    public static <K, V> DoFn<KV<K, StreamSummarySketch<ElementWrapper<V>>>,
+        KV<K, KV<V, Long>>> perKeyNonSerializable(int k) {
+      return new DoFn<KV<K, StreamSummarySketch<ElementWrapper<V>>>, KV<K, KV<V, Long>>>() {
+        @ProcessElement
+        public void processElement(ProcessContext c) {
+          KV<K, StreamSummarySketch<ElementWrapper<V>>> kv = c.element();
+          for (KV<ElementWrapper<V>, Long> pair : kv.getValue().estimateTopK(k)) {
+            c.output(KV.of(
+                kv.getKey(),
+                KV.of(pair.getKey().getElement(), pair.getValue())));
+          }
+        }
+      };
+    }
+
+    /**
+     * {@link DoFn} that retrieves the top k most frequent elements per key in a
+     * {@link PCollection} of {@link KV}s with {@link StreamSummarySketch} containing
+     * serializable objects.
+     *
+     * @param k     the top k most frequent elements to return
+     */
+    public static <K, V> DoFn<KV<K, StreamSummarySketch<V>>,
+        KV<K, KV<V, Long>>> perKeySerializable(int k) {
+      return new DoFn<KV<K, StreamSummarySketch<V>>, KV<K, KV<V, Long>>>() {
+        @ProcessElement
+        public void processElement(ProcessContext c) {
+          KV<K, StreamSummarySketch<V>> kv = c.element();
+          for (KV<V, Long> pair : kv.getValue().estimateTopK(k)) {
+            c.output(KV.of(
+                kv.getKey(),
+                KV.of(pair.getKey(), pair.getValue())));
+          }
+        }
+      };
+    }
+  }
+
+  /** Wrap StreamLib's Stream Summary sketch. */
+  public static class StreamSummarySketch<T> extends StreamSummary<T> {
+
+    public StreamSummarySketch() {
+      super();
+    }
+
+    public StreamSummarySketch(int capacity) {
+      super(capacity);
+    }
+
+    /** Merge several {@link StreamSummarySketch} into this one. */
```

Comment:
Did you mention above that this function is broken in some way? Add a comment here?

Reply:
I mentioned it in the outter class Javadoc but I can put it here two, yes.